### PR TITLE
Add flake8 linting to RHEL6 

### LINKED ISSUE #1262: Run Flake8 lint on RHEL6
Currently, flake8 is run only on RHEL7 and 8 and not on RHEL6. According to [the documentation](http://flake8.pycqa.org/en/latest/#installation) it is necessary to run flake8 with the exact Python version that is used. Thus to be sure that the syntax is ok even for the older Python version, we have to run in to RHEL6 too.

Tackled in #1251.

### PR DESCRIPTION
**Pull request by Glutexo**
Both RHEL 7 stages run _flake8_ linting, but the RHEL 6 one doesn’t. Flake8 documentation [says](http://flake8.pycqa.org/en/latest/) that it is important to run it on the exact version of Python used. Added this test from Python 2.6 used by RHEL 6.

This PR intends to replace @eduardocerqueira’s #1342 and my former #1335.

~~The reason why `flake8` command failed was because a version from RPM was used even though we wanted the one [described](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-2eeaed663bd0d25b7e608891384b7298R67) in [_setup.py_](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-2eeaed663bd0d25b7e608891384b7298). Fixed by [calling](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R25) the one installed by PIP in the user home directory directly using its path.~~

~~I didn’t introduce a virtual environment to RHEL 6 in this PR. There is already another [PR](
https://github.com/RedHatInsights/insights-core/pull/1451) for that. (#1451) I [downgraded](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-2eeaed663bd0d25b7e608891384b7298R67) Flake8 to the last version that supports Python 2.6. This version works on newer RHELs fine.~~

Thanks to #1451 we’re already [using](https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L22) virtualenv on [RHEL 6](
https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L13). That means that adding [another one](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R31) to run [Flake8](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R34) has become simple. It just must use the same [hack](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R33) as the [testing](https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L22) one.

@eduardocerqueira or @bfahr may want to review this.

Fixes #1262.

### CODE REVIEW

**Comment by Glutexo** [comment]
#1451 got merged. We’re now using a [virtual environment](https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L22) for [RHEL 6](https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L13) too. Rebased this PR so it nows creates another [virtualenv](https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R31) for [_https://github.com/RedHatInsights/insights-core/pull/1452/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R29_] linting too, the same way as the other [branches](https://github.com/RedHatInsights/insights-core/blob/05c224f1292727f40234054cdb2b40caf93801d8/Jenkinsfile#L31) do.

File: setup.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def maybe_require(pkg):
 ])
 
 linting = set([
-    'flake8==3.3.0',
+    'flake8==2.6.2',
```
[the remark this answers is not in the document]
Comment (Glutexo) [reply setup.py]: I was thinking about that too, had a condition there. Then I thought it brings unnecessary complexity. There are more other downgraded packages, some of them being more important (like [pyOpenSSL](https://github.com/RedHatInsights/insights-core/issues/1373)), that are not conditioned like this. So I concluded that I can just put it there and remove the downgrade when Red Hat stops supporting RHEL 6 with its Python 2.6.

I skimmed the [release notes](http://flake8.pycqa.org/en/latest/release-notes/) and it looks like there is not much to worry about. There was Python 3.6 support [added](http://flake8.pycqa.org/en/latest/release-notes/3.3.0.html), but our pipeline [shows](https://jenkins-insights-ci.1b13.insights.openshiftapps.com/blue/organizations/jenkins/insights-core/detail/PR-1452/13/pipeline/23) that this older version still works fine. Most of the changes are related to various options and integrations with other tools and only [few](http://flake8.pycqa.org/en/latest/release-notes/3.2.0.html) touches the actual linting.

What do you think, is it worth it to add a condition?

```python
from platform import python_version
from pkg_resources import parse_version

python26 = parse_version(python_version()) < parse_version("2.7")
flake8_version = "2.6.2" if python26 else "3.0.0"
linting = set(["flake8=={}".format(flake8_version)])
```

**Comment by YakovLittle** [comment]
test this please

**Comment by YakovLittle** [comment]
test this please

**Comment by YakovLittle** [comment]
test this please

**Comment by YakovLittle** [comment]
test this please

File: setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Glutexo) [reply setup.py]: Just pinging, @bfahr. What do you think is better: to keep the code simple with an older (but working) version of Flake8, or to use the newer one always when possible?

**Pull request merged**